### PR TITLE
Require images to have at least one tag

### LINKED ISSUE #12: Required Tag Association
#### As a user I want all images to have at least one tag.

__Acceptance criteria__:
- [ ] I cannot save an image without adding at least one tag.
- [ ] An error message is associated with the appropriate input field when a
  tag is not provided.

__Dependencies__:
- Image Tags

### PR DESCRIPTION
**Pull request by tlconnor**
Closes #12

### CODE REVIEW

**Comment by bboe** [comment]
image

File: test/controllers/images_controller_test.rb [hunk 1]
```diff
@@ -11,8 +11,8 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
   end
 
   test 'should get index with results' do
-    Image.create! url: 'http://example.image.url'
-    Image.create! url: 'http://example.second.image.url'
+    Image.create! url: 'http://example.image.url', tag_list: 'a,b,c'
```
Comment (bboe) [inline test/controllers/images_controller_test.rb]: How can you avoid needing to update all your test cases whenever you want to add a required field?

**Comment by bboe** [comment]
Aside from the deployment issue, I had only one comment pertaining to simplifying your test-instance creation.

I was unable to do QA, but based on the code, I'm not terribly worried about that part. Please merge when the deployment issue is resolved.

File: test/controllers/images_controller_test.rb (hunk 1 above)
Comment (tlconnor) [inline test/controllers/images_controller_test.rb]: @bboe I'm interested to know what your suggested change here would be.  Fixtures and factories are two common solutions.

Comment (bboe) [inline test/controllers/images_controller_test.rb]: A simple private method `create_image` factory is what I would suggest for these tests.

**Pull request merged**